packages/analytics-core/src/plugins/destination.ts

```diff
@@ -151,7 +151,12 @@ export class Destination implements DestinationPlugin {
 
     const batches = chunk(list, this.config.flushQueueSize);
 
-    await Promise.all(batches.map((batch) => this.send(batch, useRetry)));
+    // Promise.all() doesn't guarantee resolve order.
+    // Sequentially resolve to make sure backend receives events in order
+    await batches.reduce(async (promise, batch) => {
+      await promise;
+      return await this.send(batch, useRetry);
+    }, Promise.resolve());
```

Comment:
I just noticed that it's auto changed by Intellij 

I had 
```
await batches.reduce(
      (promise, batch) => {
        return promise.then(() => this.send(batch, useRetry))
      },
      Promise.resolve()
    );
```
After clicking that "convert to async function", it updated to what we have here. 
<img width="490" alt="image" src="https://github.com/user-attachments/assets/61401077-7022-4648-b8a9-55f68454262d" />

I was expecting to have
```
await batches.reduce(
      async (promise, batch) => {
        await promise;
        return this.send(batch, false);
      },
      Promise.resolve()
    );
```

It's still working although a little bit redundant. Should clean it in [AMP-120671](https://amplitude.atlassian.net/browse/AMP-120671).